blockdevice/encryption/luks/luks.go

```diff
@@ -0,0 +1,110 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package luks provides a way to call LUKS2 cryptsetup.
+package luks
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+
+	"github.com/talos-systems/go-blockdevice/blockdevice/encryption"
+	"github.com/talos-systems/go-blockdevice/blockdevice/filesystem/luks"
+	"github.com/talos-systems/go-blockdevice/blockdevice/util"
+)
+
+// LUKS implements LUKS2 encryption provider.
+type LUKS struct {
+	key    []byte
+	cipher string
+}
+
+// New creates new LUKS2 encryption provider.
+func New(key []byte, cipher string) *LUKS {
+	return &LUKS{
+		key:    key,
+		cipher: cipher,
+	}
+}
+
+// Open runs luksOpen on a device and returns mapped device path.
+func (l *LUKS) Open(deviceName string, key []byte) (string, error) {
```

Comment:
I got rid of l.key everywhere.
I think in case if we want to try opening the partition with different keys, it's more convenient to have `key` as an argument, instead of a struct field.